lua/entities/gmod_wire_egp/lib/egplib/transmitreceive.lua

```diff
@@ -684,7 +675,11 @@ else
 			LocalPlayer():ChatPrint("[EGP] Received EGP object reload. " .. decoded.IsLastScreen .. " screens' objects were reloaded.")
 		end
 	end
+
 	net.Receive("EGP_Request_Transmit", function(len,ply)
-		EGP:ReceiveDataStream(net.ReadTable())
+		local amount = net.ReadUInt(16)
+		local data = net.ReadData(amount)
+		local tbl = WireLib.von.deserialize(util.Decompress(data))
```

Comment:
Decompress needs a max length specified.

Reply:
The second argument is optional, it works fine without
Is there a reason you want a specific max length? And if so what should it be?

Reply:
Yes, it prevents zip bombing. It should match a max length prior to compression.

Reply:
@wrefgtzweve

Reply:
I don't think a server -> client zip bomb is a big threat, sorry for the slow response I've been busy the past few days.

Reply:
These are client controlled streams though. If someone figures out a way to get 100000000 egp objects to compress down to a few KB then yeah it's a threat.